doc/regulative.md

```diff
@@ -0,0 +1,451 @@
+are assessed, the way the bill is calculated at the end of the meal.
+
+The notepad relays the order from the patron via the waiter to the bill.
+
+In L4, the State mechanism relays information about offences to the penalty section. We `put` information into state variables, and `get` them out later.
+
+## Object-Level versus Assertion-Level
+
+L4 uses a state transition formalism to represent the moving parts of
+a regulative rule.
+
+Whenever somebody has to do something -- or refrain from doing
+something -- within a certain time period, and face consequences for
+noncompliance, we use L4's regulative statements to express those
+rules.
+
+That formalism _could_ use the bloodless, mechanical form of the
+rules: one thing leads to another and another and another. We could
+render a finite state automaton, or a state transition system, using
+the most unopinionated, nonjudgemental language imaginable. This is
+the "object level" representation of a normative system.
+
+"If this action is not taken, a penalty fee will be added to the invoice."
+
+"If the penalty fee is not paid, borrowing privileges will be suspended."
+
+Maybe a library user doesn't care if they lose borrowing privileges
+forever. Then the "you must pay a fine" is a toothless rule: it is not
+strongly _enforceable_.
+
+## Deontic Statements are Bounded
+
+The statement "you must pay a fine" carries with it an unspoken
+complement: "or else you will not be allowed to borrow again."
+
+These complements usually exist in informal discourse, yet they usually go unspoken.
+
+When you hear a "must" but don't hear the explicit consequence, the
+consequence is frequently "or you commit an offence, and are subject
+to some sort of legal penalty."
+
+Or it might be "or you sin against your fellows, and are subject to
+social misapprobation."
+
+"You must pick up your child from childcare before 7pm."
+
+In L4, every deontic statement is bounded. Even a "may" permission is
+bounded, in that taking that course of action that eventually causes
+some other party to assume some obligation that they would otherwise
+not.
+
+## Deontic consequences are explicated at the Assertion Level
+
+"You must do this thing if you don't want to pay a monetary penalty"
+
+"There is no way to reach a FULFILLED outcome without paying a penalty, other than by doing this thing."
+
+These kinds of statements are formalized in LTL / CTL as assertions,
+or properties, _over_ the object level of the state transition system.
+
+## We can treat `MUST`, `MAY`, and `SHANT` as sugar over a simple `DO`
+
+The purely mechanistic object-level form of a regulative rule is structured like this:
+
+```
+    §  clause 1
+   IF  preconditions
+ UPON  trigger event  -- only used at top-level, otherwise this clause follows from some other clause
+PARTY  p
+   DO  action
+       with  certain criteria
+	   to    some target
+HENCE  clause 2 (... AND clause 4 AND clause 6 OR clause 8)
```

Comment:
As noted in https://github.com/smucclaw/lam4/blob/88b52216a48a14bcd819aab953da59241b2dffa7/docs/proposal_normative_constructs_aug26_2024.md?plain=1#L218
I'm not sure that the HENCE ... LEST would allow us to express certain things that we would like to express. I think Matt Waddington talked about similar things at the last call as well, though I might have misunderstood what he said; at any rate, the syntax proposed in that link probably is closer to the kind of 'if ... then ...' syntax that Matt had been talking about.

But of course, there might be other reasons for preferring HENCE LEST.

Reply:
Nice, thanks for re-posting this. I remember seeing it go by previously and thinking we should discuss it together as a group when the time was right. Now the time is right!

Reply:
Matt mentioned he didn't see a requirement for representing a multiplicity of jurisdictions within the same document. However the more general notion of separation – of the commitment of an offence from the penalty that it attracts – remains valid, so I would like to preserve that notion in any future modified form of your proposal.

Reply:
I think the functionality described in https://github.com/smucclaw/lam4/blob/88b52216a48a14bcd819aab953da59241b2dffa7/docs/proposal_normative_constructs_aug26_2024.md?plain=1#L218 is now accommodated in the latest regulative-proposal doc so let's go with that.

I forgot to mention an additional point – I wanted the coupling between THEN/ELSE and HENCE/LEST to be explicit. I was worried about a dangling ELSE in the same way Haskell insisted that every IF/THEN have an ELSE.

In next round of work I want to try implementing the offence/offence/offence/penalty pattern in the latest syntax so we can see how we feel about it.